net/net_demo_installer

```diff
@@ -0,0 +1,212 @@
+echo "\
+[defaults]
+host_key_checking = False" > ~/.ansible.cfg
+
+echo "Contents of $HOST_INVENTORY_FILE:"
+echo "-----------------"
+cat $HOST_INVENTORY_FILE
+echo "-----------------"
+
+# Verify ansible can reach all hosts
+echo "Verifying ansible reachability"
+ansible all -i $HOST_INVENTORY_FILE -a "/bin/cat /etc/issue.net" >& $HOST_LOG_FILE
+grep FAIL $HOST_LOG_FILE >& /dev/null
+if [ $? -eq 0 ]; then
+   echo "WARNING"
+   echo "WARNING Some of the hosts are not accessible via passwordless SSH"
+   echo "WARNING"
+   echo "`grep FAIL $HOST_LOG_FILE`"
+   echo " "
+   echo "This means either the host is unreachable or passwordless SSH is not"
+   echo "set up for it. It is RECOMMENDED that you resolve this before"
+   echo "proceeding. If this is JUST a passwordless SSH issue, you may choose"
+   echo "to proceed and respond to password prompts(there would a multitude"
+   echo "of those...)"
+
+   ConfirmPrompt
+   ans_opts=k
+fi
+
+# Verify Ubuntu version on all hosts
+echo "Verifying Ubuntu versions on all hosts..."
+cat $HOST_LOG_FILE |  awk '/Ubuntu/ {x=2 ; if (substr($x,1,2) != 15) { print "ERROR Some of the hosts need to be upgraded to Ubuntu 15"; exit 1} }'
+if [ $? -eq 1 ]; then
+cat $HOST_LOG_FILE
+exit 1
+fi
+
+echo "Verifying sudo access on all hosts..."
+ansible all -i $HOST_INVENTORY_FILE -a "sudo -n /bin/cat /etc/sudoers" >& $HOST_SUDO_LOG_FILE 
+res=`grep FAIL $HOST_SUDO_LOG_FILE`
+if [ ! -z "$res" ]; then
+   echo "WARNING Some of the hosts do not have passwordless sudo enabled"
+   echo "WARNING"
+   echo "`grep FAIL $HOST_SUDO_LOG_FILE`"
+   echo " "
+   echo "It is RECOMMENDED that you resolve this before proceeding. However,"
+   echo "you may choose to proceed and respond to password prompts (there "
+   echo "could be a multitude of those...)"
+   echo " "
+
+   ConfirmPrompt
+   ans_opts=K$ans_opts
+fi
+
+if [ ! -z "$ans_opts" ]; then
+   ans_opts=-$ans_opts
+fi
+
+echo ""
+echo "Ansible access and Ubuntu version verification completed on all hosts"
+set -e
+
+echo "-----------------"
+
+# Get IP of all nodes
+rm -rf $HOST_IP_LOG_FILE
+ansible $NETPLUGIN_HOSTGROUP -m setup $ans_opts -i $HOST_INVENTORY_FILE | grep -A 4 ansible_eth1 | grep address | awk -F \" '{print $4}' >> $HOST_IP_LOG_FILE
+# Use netmaster's default ip as service vip
+svc_vip=$(ansible node1 -m setup $ans_opts -i $HOST_INVENTORY_FILE | grep -A 4 ansible_eth1 | grep address | awk -F \" '{print $4}')
+no_proxy=$(sed ':a;N;$!ba;s/\n/,/g' $HOST_IP_LOG_FILE)",127.0.0.1,localhost,netmaster"
+echo '{"env":{"http_proxy":"'$http_proxy'", "HTTP_PROXY":"'$http_proxy'", "https_proxy":"'$https_proxy'", "no_proxy":"'$no_proxy'"}, "monitor_interface": "eth1", "etcd_peers_group": '"\"$NETPLUGIN_HOSTGROUP\""', "service_vip": '"\"$svc_vip\""'}' > $ENV_FILE
+    ansible-playbook $ans_opts -i $HOST_INVENTORY_FILE  -e "`cat $ENV_FILE`" $ANSIBLE_SITE_FILE
+
+# Finally run restart to clean up any left over state.
+echo "Waiting 45 sec for services to initialize..."
```

Comment:
Do we need this? This for ACI was done to keep the ACI gateway up and running? Perhaps reduce it to 5 secs...